sdk/eventhub/event-hubs/test/receiver.spec.ts

```diff
@@ -292,6 +292,126 @@ describe("EventHub Receiver", function(): void {
     });
   });
 
+  describe("getIterator", function(): void {
+    it("should receive messages correctly", async function(): Promise<void> {
+      const partitionId = partitionIds[0];
+
+      const time = Date.now();
+      // ensure messages exist to retrieve
+      const messageCount = 5;
+      const sentEventData: EventData[] = [];
+      for (let i = 0; i < messageCount; i++) {
+        sentEventData.push({
+          body: `getIterator test message ${Date.now()} - ${i}`
+        });
+      }
+
+      const sender = client.createSender({partitionId: partitionId});
+      try {
+        await sender.send(sentEventData);
+      } finally {
+        await sender.close();
+      }
+
+      receiver = client.createReceiver(partitionId, {
+        eventPosition: EventPosition.fromEnqueuedTime(time)
+      });
+      const eventIterator = receiver.getEventIterator();
+
+      let messagesReceivedCount = 0;
+      const data: ReceivedEventData[] = [];
+      for await (const event of eventIterator) {
+        data.push(event);
+        if (++messagesReceivedCount >= messageCount) {
+          break;
+        }
+      }
+
+      debug("received messages: ", data);
+      data.length.should.equal(messageCount, `Failed to receive ${messageCount} expected messages`);
+    });
+  });
+
+  describe("mix and match receive methods", function(): void {
+    it("should maintain sequence across calls", function(done: Mocha.Done): void {
+      // wrap test to allow mixing async/await with done
```

Comment:
I believe you had to do this because the only way to exit the test from inside the callback to the streaming receiver was by calling done()?

Reply:
Yes, that's correct.